lib/saml_idp/encryptor.rb

```diff
@@ -9,47 +9,43 @@ class Encryptor
     def initialize(opts)
       self.block_encryption = opts[:block_encryption]
       self.key_transport = opts[:key_transport]
-      self.cert = opts[:cert]
+      self.cert = opts[:cert].to_s.gsub(/-+(BEGIN|END) CERTIFICATE-+/, '')
     end
 
-    def encrypt(raw_xml) 
+    def encrypt(raw_xml)
       encryption_template = Nokogiri::XML::Document.parse(build_encryption_template).root
       encrypted_data = Xmlenc::EncryptedData.new(encryption_template)
       @encryption_key = encrypted_data.encrypt(raw_xml)
       encrypted_key_node = encrypted_data.node.at_xpath(
         '//xenc:EncryptedData/ds:KeyInfo/xenc:EncryptedKey',
         Xmlenc::NAMESPACES
-      )   
+      )
       encrypted_key = Xmlenc::EncryptedKey.new(encrypted_key_node)
       encrypted_key.encrypt(openssl_cert.public_key, encryption_key)
       xml = Builder::XmlMarkup.new
       xml.EncryptedAssertion xmlns: Saml::XML::Namespaces::ASSERTION do |enc_assert|
         enc_assert << encrypted_data.node.to_s
-      end 
+      end
     end
 
     def openssl_cert
-      if cert.is_a?(String)
-        @_openssl_cert ||= OpenSSL::X509::Certificate.new(Base64.decode64(cert))
-      else
-        @_openssl_cert ||= cert
-      end 
-    end 
+      @_openssl_cert ||= OpenSSL::X509::Certificate.new(Base64.decode64(cert))
```

Comment:
What's reason for this change. I think it might break that some peoples handling certificate object by themselves.